db/db.go

```diff
@@ -265,7 +265,15 @@ func IsConnectionError(err error) bool {
 	if err == nil {
 		return false
 	}
-	lowerCaseError := strings.ToLower(err.Error())
+
+	// The new driver stringifies command errors as "(Name) Message" rather than just "message". Cast to the
+	// CommandError type if possible to extract the correct error message.
+	errMsg := err.Error()
+	if cmdErr, ok := err.(mongo.CommandError); ok {
```

Comment:
A CommandError in general is supposed to be a type of error returned by the server (complete with a code and message both set by the server). Now, I'm not sure how the go driver is using them, but if their `CommandError` is defined along those lines, a connection related error would never show up as a command error because if there's no connection to the server, it can't return an error back to the client.

Reply:
The specific case I was targeting was "not master" which is an error from the server that we consider a connection error in this code.

Reply:
Oh, I see. the "connection" is referring to a wider definition than just connecting to the server. It includes nodes not being able to connect to each other or the client not being able to connect to the primary specifically and other errors like that. Perhaps it could have been named or documented a bit better.